src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -42,7 +44,12 @@ protected override IMethodNode CreateUnboxingStubNode(MethodDesc method)
 
         public override void AttachToDependencyGraph(DependencyAnalyzerBase<NodeFactory> graph)
         {
+            CoreCLRReadyToRunHeader = new CoreCLRReadyToRunHeaderNode(Target);
+            graph.AddRoot(CoreCLRReadyToRunHeader, "ReadyToRunHeader is always generated");
 
+            var compilerIdentifierNode = new CompilerIdentifierNode();
+            CoreCLRReadyToRunHeader.Add(Internal.Runtime.ReadyToRunSectionType.CompilerIdentifier, compilerIdentifierNode, compilerIdentifierNode);
+            graph.AddRoot(compilerIdentifierNode, "CompilerIdentifierNode is always generated");
```

Comment:
The `compilerIdentifierNode` will be reported as a dependency of the ready to run header. We shouldn't need to root it.

Reply:
You're right - removed